File: app/src/main/java/com/android/learn/mcontract/RegisterLoginContract.java
package com.android.learn.mcontract;

import com.android.learn.base.mmodel.RegisterLoginData;

/**
 * Created by gaolei on 2018/6/18.
 */

public class RegisterLoginContract {

    public interface Presenter {

        void login(String account, String password);

        void register(String account, String password, String rePassword);

    }

    public interface View {

        void showRegisterResData(RegisterLoginData registerResData);

        void showLoginResData(RegisterLoginData loginData);
    }
}

File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/RegisterLoginContract.kt
package com.android.learn.mcontract

import com.android.learn.base.mmodel.RegisterLoginData
import com.android.learn.base.mview.BaseView

/**
 * Created by gaolei on 2018/6/18.
 */

class RegisterLoginContract {

    interface Presenter {

        fun login(account: String, password: String)

        fun register(account: String, password: String, rePassword: String)

    }

    interface View : BaseView {

        fun showRegisterResData(registerResData: RegisterLoginData)

        fun showLoginResData(loginData: RegisterLoginData)
    }
}
